### ISSUE TITLE
Jio TV not working

### ISSUE THREAD

**Issue by RavitejaLam** [botallen/repository.botallen#163]
**Sony TV**
- Add-on name: Jiotv
- Add-on version: 2.0.0
- Kodi version: 19.1
- Operating system version/name: Android TV

**Describe the bug**
Playback Failed - One or more items failed to play. 

**To Reproduce**
1. Go to 'Add-ons'
2. Click on 'JioTV'
3. Click on 'Genres'
4. Click on 'Entertainment'
5. Click on 'Colors HD'
4. See an error

**Expected behavior**
Colors HD should be playing

**Screenshots**
![image](https://user-images.githubusercontent.com/22323518/127818672-7f0c32df-3528-419e-886f-d5efbf443d8f.png)

**Debug log**
The debug log can be found from this link: [kodi.log](https://github.com/botallen/repository.botallen/files/6915062/kodi.log)

**Comment by iMro0t** [comment]
Are you accessing from India or outside India? By looking at logs the error is only possible when you try to access it from outside India

**Comment by pranavcool007** [comment]
same issue not able to play any channel

**Comment by RavitejaLam** [comment]
> Are you accessing from India or outside India? By looking at logs the error is only possible when you try to access it from outside India 

No, I'm accessing from india. I was able to use with old kodi version (19.0) and not latest (19.1) may be related to that.

**Comment by aslamdoctor** [comment]
Same issue with me.

**Comment by rahul260792** [comment]
Same issue with me.

**Comment by emilioescobargaviria** [comment]
same issue with me also is not the problem with the verison

**Comment by emilioescobargaviria** [comment]
same issue with me but its is not the problem with the version

**Comment by Peeeenut** [comment]
I have the same issue, accessing from India, latest Kodi. Nothing works on the add-on. Tried wiping everything and reinstalling from scratch but to no avail.

**Comment by kushraj** [comment]
where did you get kodi v19.0 ? no such version in arch repos archive

**Comment by RavitejaLam** [comment]
> where did you get kodi v19.0 ? no such version in arch repos archive

https://www.apkmirror.com/apk/xbmc-foundation/kodi/kodi-19-0-release/

**Comment by kushraj** [comment]
Oh I thought you were talking about pc version, not much use case for Android for me, as you can use the app itself.

**Comment by Peeeenut** [comment]
> where did you get kodi v19.0 ? no such version in arch repos archive

On arch you could probably go for flatpak or snap? Partial updates are unsupported so installing via pacman doesn't sound like a good idea anyway.

**Comment by kushraj** [comment]
tried the flatpak version, its still the same issue

```

2021-08-10 17:22:43.943 T:565      INFO <general>: CPythonInvoker(7, /home/kushy/.var/app/tv.kodi.Kodi/data/addons/plugin.video.jiotv/addon.py): script successfully run
2021-08-10 17:22:43.966 T:7        INFO <general>: VideoPlayer::OpenFile: plugin://plugin.video.jiotv/resources/lib/main/play/?_pickle_=8004952c000000000000007d94288c075f7469746c655f948c0b536f6e7920536978204844948c0a6368616e6e656c5f6964944b9b752e
2021-08-10 17:22:44.765 T:648      INFO <general>: Creating InputStream
2021-08-10 17:22:44.765 T:648     ERROR <general>: IAddonInstanceHandler::CreateInstance: inputstream.adaptive returned bad status "Permanent failure" during instance creation
2021-08-10 17:22:44.765 T:648     ERROR <general>: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.jiotv/resources/lib/main/play/?_pickle_=8004952c000000000000007d94288c075f7469746c655f948c0b536f6e7920536978204844948c0a6368616e6e656c5f6964944b9b752e]
2021-08-10 17:22:44.765 T:648      INFO <general>: CVideoPlayer::OnExit()
2021-08-10 17:22:44.877 T:7        INFO <general>: CVideoPlayer::CloseFile()
2021-08-10 17:22:44.877 T:7        INFO <general>: VideoPlayer: waiting for threads to exit
2021-08-10 17:22:44.877 T:7        INFO <general>: VideoPlayer: finished waiting

```

**Comment by meehatpa** [comment]
I also have the same issue.

**Comment by PrabakaranSK** [comment]
I also have the same issue (Android 8.0.0 + latest Kodi + this addon [says "Playback failed" like this thread's screenshot])

**Comment by mayankwadhwani** [comment]
Hi All, this issue can be resolved by installing the inputstream.adaptive and inputstream rtmp plugins. @iMro0t Confirmed that it is not working on multiple devices due to missing plugins. Either adding as dependency or adding in instructions should make it work.

**Comment by kedarbhise** [comment]
Thanks @mayankwadhwani.

I am facing a different issue #191 for the Disney+ Hotstar add on due to the same underlying cause of missing inputstream.adaptive and inputstream rtmp plugins.

These are added to the Disney+ Hotstar as dependencies, but still cannot install.

@iMro0t How do we manually install these plugins on Kodi?

**Comment by kushraj** [comment]
getting these after installing both plugins
```
ERROR <general>: IAddonInstanceHandler::CreateInstance: inputstream.adaptive returned bad status "Permanent failure" during instance creation
2021-08-30 13:02:21.652 T:37033   ERROR <general>: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.jiotv/resources/lib/main/play/?_pickle_=8004952c000000000000007d94288c075f7469746c655f948c0b536f6e7920536978204844948c0a6368616e6e656c5f6964944b9b752e]
```

**Comment by deepeshj** [comment]
I am facing same issue. 18.9 jio tv add-on works. In 19.1 not able to play any streams

**Comment by obitoo7** [comment]
Are you guys downloaded simple IPTV addon from Kodi repo

**Comment by chamaraju** [comment]
Yes, downloaded simple IPTV addon from Kodi repo, but not working. It showing the list of channels and none playing.

**Comment by 29Unknown29** [comment]
getting similar issue where channels are showing but not playing. tried reinstalling also installed simple iptv

**Comment by chamaraju** [comment]
open my-addons and jio tv settings and clear cache. refer here on how to: https://www.youtube.com/watch?v=Yfv1V5QLZ1E

**Comment by Shahbaz03** [comment]
I am trying this from outside India, and getting the same issue. Have tried almost everything but no luck. Does it not work outside India?

**Comment by ruchirpatel15** [comment]
> I am trying this from outside India, and getting the same issue. Have tried almost everything but no luck. Does it not work outside India?

I am also having same issues did you get any solutions?

**Comment by DirectiveTaurus** [comment]
You just simply have to turn of the extra hd channels option and then restart the app

**Comment by mosesjebasingh** [comment]
> You just simply have to turn of the extra hd channels option and then restart the app

Thank you very much. This worked for me.
